src/module/system/schema-data-fields.ts

```diff
@@ -100,4 +100,28 @@ class PredicateField<
     }
 }
 
-export { LaxSchemaField, PredicateField, SlugField };
+class ItemAlterationValueField extends fields.DataField<
+    string | number | null,
+    string | number | null,
+    true,
+    true,
+    true
+> {
+    static override get _defaults() {
+        return mergeObject(super._defaults, {
+            required: true,
+            nullable: true,
+            initial: null,
+        });
+    }
+
+    protected _cast(value: unknown): string | number | null {
+        return typeof value === "string" || typeof value === "number" || value === null ? value : null;
```

Comment:
```suggestion
        return value;
```

Reply:
Shouldn't it cast an invalid value to valid one?

Reply:
No, it should if the value isn't valid.

Reply:
DataField:
```js
  /**
   * Cast a non-default value to ensure it is the correct type for the field
   * @param {*} value       The provided non-default value
   * @returns {*}           The standardized value
   * @protected
   */
  _cast(value) {
    throw new Error(`Subclasses of DataField must implement the _cast method`);
  }
```
NumberField:
```js
  /** @override */
  _cast(value) {
    return Number(value);
  }
  ```

Reply:
Yeah, that's not great since a `string` or `ActorPF2e` isn't a number. A `NumberField` shouldn't take anything in the entire world and coerce it into one.